File: tl/src/main/java/com/github/badoualy/telegram/tl/api/TLAbsFileLocation.java

package com.github.badoualy.telegram.tl.api;


import com.github.badoualy.telegram.tl.core.TLObject;



public abstract class TLAbsFileLocation extends TLObject {

    protected long volumeId;

    protected int localId;

    protected long secret;


    public long getVolumeId() {
        return volumeId;
    }

    public void setVolumeId(long value) {
        this.volumeId = value;
    }

    public int getLocalId() {
        return localId;
    }

    public void setLocalId(int value) {
        this.localId = value;
    }

    public long getSecret() {
        return secret;
    }

    public void setSecret(long value) {
        this.secret = value;
    }

}

File: tl/src/main/kotlin/com/github/badoualy/telegram/tl/api/TLAbsFileLocation.kt
package com.github.badoualy.telegram.tl.api

import com.github.badoualy.telegram.tl.core.TLObject

/**
 * Abstraction level for the following constructors:
 * * [fileLocation#53d69076][TLFileLocation]
 * * [fileLocationUnavailable#7c596b46][TLFileLocationUnavailable]
 *
 * @author Yannick Badoual yann.badoual@gmail.com
 * @see <a href="http://github.com/badoualy/kotlogram">http://github.com/badoualy/kotlogram</a>
 */
abstract class TLAbsFileLocation : TLObject() {
    abstract var volumeId: Long

    abstract var localId: Int

    abstract var secret: Long
}
